src/api/include/migraphx/migraphx.hpp

```diff
@@ -344,7 +344,10 @@ struct program_parameter_shapes : MIGRAPHX_HANDLE_BASE(program_parameter_shapes)
     std::vector<const char*> names() const
     {
         std::vector<const char*> result(this->size());
-        call(&migraphx_program_parameter_shapes_names, result.data(), this->get_handle_ptr());
+        if(result.size() > 0)
+        {
+            call(&migraphx_program_parameter_shapes_names, result.data(), this->get_handle_ptr());
```

Comment:
Why is there a seg fault? Is there a way to fix this in the [`get_names`](https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/blob/develop/tools/api/api.cpp#L102) function instead?

Reply:
I think if the size of result is 0, result.data() is nullptr, and then it failed.

Reply:
The check if (out == nullptr) caused the problem. It has nothing to do with the get_name function. Is there a way to remove the check?

Reply:
Let me look into this.

Reply:
I see, its not a seg fault, there is an exception thrown because the c api returns an error. I think changing here in the C++ api is acceptable.